wkcuber/utils.py

```diff
@@ -74,48 +77,66 @@ def get_regular_chunks(min_z, max_z, chunk_size):
         i += chunk_size
 
 
-def add_jobs_flag(parser):
+def add_distribution_flags(parser):
     parser.add_argument(
-        "--jobs", "-j", help="Parallel jobs", type=int, default=cpu_count()
+        "--jobs",
+        "-j",
+        default=cpu_count(),
+        type=int,
+        help="Number of processes to be spawned.",
     )
 
+    parser.add_argument(
+        "--distribution_strategy",
+        default="multiprocessing",
+        choices=["slurm", "multiprocessing"],
+        help="Strategy to distribute the task across CPUs or nodes.",
+    )
 
-def pool_init(lock):
-    global process_pool_lock
-    process_pool_lock = lock
+    parser.add_argument(
+        "--job_resources",
+        default=None,
+        help='Necessary when using slurm as distribution strategy. Should be a JSON string (e.g., --job_resources=\'{"mem": "10M"}\')',
+    )
 
 
-def pool_get_lock():
-    global process_pool_lock
-    try:
-        return process_pool_lock
-    except NameError:
-        return None
+def get_executor_for_args(args):
+    if args is None:
+        # For backwards compatibility with code from other packages
+        # we allow args to be None. In this case we are defaulting
+        # to these values:
+        args.distribution_strategy = "multiprocessing"
+        args.jobs = cpu_count()
 
+    executor = None
 
-class ParallelExecutor:
-    def __init__(self, jobs):
-        self.lock = Lock()
-        if python_version() >= "3.7.0":
-            self.exec = ProcessPoolExecutor(
-                jobs, initializer=pool_init, initargs=(self.lock,)
+    if args.distribution_strategy == "multiprocessing":
+        if args.jobs is None:
```

Comment:
Is this mandatory or could we use a sensible default (e.g. num of cpus)?

Reply:
Good point, the default is already num of cpus, so there is no sense in checking for None here. I removed the code.